### ISSUE TITLE
meta(fuzzer): tracking issue for improvements to property tests and invariant tests

### ISSUE THREAD

**Issue by mds1** [foundry-rs/foundry#4433]
### Component

Forge

### Describe the feature you would like

This issue tracks general fuzzer improvements that improve both property tests and invariant tests. (There will be a separate tracking issue for things that are strictly invariant test related).

Items are roughly ordered by my opinion of priority, both at the header-level and the bullet-level. I say "roughly ordered" because some items are much bigger scope than others so having a strict ordering doesn't make much sense.

## Fuzzer Benchmark/Tests

First priority is some tests and benchmarks. to verify behavior of the fuzzer has not regressed as it's changed. Potential test cases:

- https://github.com/foundry-rs/foundry/issues/2851
- https://github.com/crytic/echidna/tree/master/tests/solidity
- https://blog.trailofbits.com/2023/02/27/reusable-properties-ethereum-contracts-echidna/ 

This is tracked in a separate issue here: https://github.com/foundry-rs/foundry/issues/3411

## Refactor / Tech Debt / Cleanup

- Remove proptest? It has some limitations and isn't currently maintained (ref discussions in https://github.com/foundry-rs/foundry/issues/990 and https://github.com/proptest-rs/proptest/issues/268). This would be a big change so shouldn't be done until we have tests/benchmarks in place.
- https://github.com/foundry-rs/foundry/issues/3153

## Features / Performance

High Priority
- https://github.com/foundry-rs/foundry/issues/3062 OR https://github.com/foundry-rs/foundry/issues/4085 — Before implementing, need to decide if we like the config file approach or a comment-based approach, then finalize UX, then implement. Comment approach does add flexibility and allows future expansion to things like #4097
- https://github.com/foundry-rs/foundry/issues/3521
- https://github.com/foundry-rs/foundry/issues/585

Low priority:
- https://github.com/foundry-rs/foundry/issues/4867
- https://github.com/foundry-rs/foundry/issues/990 — Not supported by proptest so currently hard, but may be easy after a refactor. Low priority because it can be replicated by setting runs to max uint
- https://github.com/foundry-rs/foundry/issues/4090
- https://github.com/foundry-rs/foundry/issues/4097 — If we go with the comment approach for [#3062](https://github.com/foundry-rs/foundry/issues/3062), this is doable
- https://github.com/foundry-rs/foundry/issues/3844 — I think this would be hard to implement with the current architecture since things run in parallel, and you can accomplish the same thing with `vm.writeLine`, so this isn't necessary IMO. May be worth closing
- https://github.com/foundry-rs/foundry/issues/4091 — This flag would conflict with the existing `max_test_rejects` flag, so one needs to take precedence and we need warn if the both exist. Personally this feels like the complexity/config growth isn't worth it since the same functionality can already be accomplished, so may be worth closing

## Failure Persistence and Replay

I have this last because the functionality can be replicated a bit tediously by copying failed tests into concrete tests. 

- https://github.com/foundry-rs/foundry/issues/2551
- https://github.com/foundry-rs/foundry/issues/2552

## Long Term Fuzz Techniques

Some more advanced techniques to consider down the road:
- Using outputs from symbolic execution to seed the fuzzer
- Techniques described in the discussion in https://github.com/foundry-rs/foundry/issues/3277

**Comment by zerosnacks** [comment]
Closing in favor of https://github.com/foundry-rs/foundry/issues/8076

Made sure that all unresolved tickets have been incorporated (this was already the case) with the exception of #4090, this was moved to the cheatcodes meta: https://github.com/foundry-rs/foundry/issues/4439
